src/components/engagement/engagement.service.ts

```diff
@@ -15,8 +26,274 @@ import {
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditCommunicationsCompleteDate:ACL {canEditCommunicationsCompleteDate: true})-[:toNode]->(languageEngagement)
+
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadModifiedAt:ACL {canReadModifiedAt: true})-[:toNode]->(languageEngagement)-[:modifiedAt {active: true}]->(modifiedAt:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditModifiedAt:ACL {canEditModifiedAt: true})-[:toNode]->(languageEngagement)
+        
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadStartDate:ACL {canReadStartDate: true})-[:toNode]->(languageEngagement)-[:startDate {active: true}]->(startDate:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditStartDate:ACL {canEditStartDate: true})-[:toNode]->(languageEngagement)
+
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadEndDate:ACL {canReadEndDate: true})-[:toNode]->(languageEngagement)-[:endDate {active: true}]->(endDate:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditEndDate:ACL {canEditEndDate: true})-[:toNode]->(languageEngagement)
+
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadInitialEndDate:ACL {canReadInitialEndDate: true})-[:toNode]->(languageEngagement)-[:initialEndDate {active: true}]->(initialEndDate:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditInitialEndDate:ACL {canEditInitialEndDate: true})-[:toNode]->(languageEngagement)
+
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadLastSuspendedAt:ACL {canReadLastSuspendedAt: true})-[:toNode]->(languageEngagement)-[:lastSuspendedAt {active: true}]->(lastSuspendedAt:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditLastSuspendedAt:ACL {canEditLastSuspendedAt: true})-[:toNode]->(languageEngagement)
+
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadLastReactivatedAt:ACL {canReadLastReactivatedAt: true})-[:toNode]->(languageEngagement)-[:lastReactivatedAt {active: true}]->(lastReactivatedAt:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditLastReactivatedAt:ACL {canEditLastReactivatedAt: true})-[:toNode]->(languageEngagement)
+
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canReadStatusModifiedAt:ACL {canReadStatusModifiedAt: true})-[:toNode]->(languageEngagement)-[:statusModifiedAt {active: true}]->(statusModifiedAt:Property {active: true})
+        WITH * OPTIONAL MATCH (requestingUser)<-[:member]-(canEditStatusModifiedAt:ACL {canEditStatusModifiedAt: true})-[:toNode]->(languageEngagement)
```

Comment:
- initialEndDate
- lastSuspendedAt
- lastReactivatedAt
- statusModifiedAt
- modifiedAt
Are all calculated fields. The user will never be able to write them directly. So they don't need the canEdit* checks.
Also it might make sense if the canRead* were just from the status field since those are dates based on that.